### PR TITLE
DATAUP-288 left column prep - factory pattern

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

Convert the base `bulkImportCell` module to use the factory pattern.

It really only needs to expose the `make` function, other items are added to the factory result for help with testing and cell deletion if needed later.

This is on the same branch as other changes to #1954 , so there's some overlap until that gets merged, but I wanted to get this out there.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
Really there should be no change - fire up the narrative and try out a bulk import cell, and run tests. Tests should all have been updated accordingly.
- [x] Tests pass in travis and locally 
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: nbextensions/bulkImportCell/cellTabs.js [hunk 1]
```diff
@@ -28,41 +34,46 @@ define([
...
          *       }
          */
-        constructor(options) {
-            this.bus = options.bus;
-            this.ui = options.ui;
-            this.tabToggleAction = options.toggleAction;
-            this.controlBarTabs = options.tabs;
-        }
+        let bus = options.bus,
+            ui,
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/cellTabs.js:38]: will this always be created by the `start` method, and never be passed in? Method docs should be updated as they still mention `ui`.

File: nbextensions/bulkImportCell/cellControlPanel.js [hunk 2]
```diff
@@ -13,82 +13,80 @@ define([
...
-                bus: this.bus,
-                toggleAction: options.tabs.toggleAction,
-                tabs: options.tabs.tabs
-            });
+        // this.cellTabs = new CellTabs({
+        //     ui: this.ui,
+        //     bus: this.bus,
+        //     toggleAction: options.tabs.toggleAction,
+        //     tabs: options.tabs.tabs
+        // });
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/cellControlPanel.js:55]: Does this still need to be here if it's called from `bulkImportCell.js`?

File: nbextensions/bulkImportCell/cellTabs.js (hunk 1 above)
Comment (briehl) [reply nbextensions/bulkImportCell/cellTabs.js:37]: Yup, fixed the docs, and rearranged where they go (above the function as usual).

File: nbextensions/bulkImportCell/cellControlPanel.js (hunk 2 above)
Comment (briehl) [reply nbextensions/bulkImportCell/cellControlPanel.js]: Nope, removed all references to tabs from the control panel.

**Comment by briehl** [comment]
Whew, all 8 checks pass!

**Review by ialarmedalien — APPROVED** [review]
Looks good!

**Pull request merged**